### ISSUE TITLE
Cant import

### ISSUE THREAD

**Issue by Daemon-ser** [yuce/pyswip#48]
I got a problem when using ‘pyswip'.  Maybe the  function re.search miss an argument(‘swi_ver’) ? Is it  necessary to input the version of swi-prolog? It never happened to me before.  
![image](https://user-images.githubusercontent.com/22999510/54805895-575bc600-4cb3-11e9-83f1-5a54e52752b7.png)
![image](https://user-images.githubusercontent.com/22999510/54805909-6d698680-4cb3-11e9-9896-afb9392eb406.png)

**Comment by Freakwill** [comment]
same problem. it must be a bug.

**Comment by zhanwenchen** [comment]
Same problem. The code here makes no sense because `re.search(regex_pattern, string_to_search)` takes two arguments but on line 310 here there is only the `pattern` argument, so of course the code raises.

**Comment by zhanwenchen** [comment]
@yuce After some digging, I think a deeper problem is that `pyswip` doesn't support SWI-Prolog 8.x, because the version-seeking functions in `pyswip/core.py` look for `libswipl.xx` and `libpl.xx` but those no longer exist starting with swipl 8.0.1. See [the official list of all stable releases](http://www.swi-prolog.org/download/stable?show=all).

**Comment by zhanwenchen** [comment]
I finally found a solution: the only SWI-Prolog version that has worked for me is Homebrew's `swi-prolog` 7.6.4 (latest 7.x.x stable release). Because Homebrew does not keep track of different versions of `swi-prolog`, I had to really dig to find the old 7.6.4 "formula". Now it works with the current version of pyswip (0.2.8).

TL; DR: Remove your current SWI-Prolog install (and remove any associated PATH editing) and `brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/09a94009142a6265b0b8e322463100610aeda964/Formula/swi-prolog.rb` and call it a day.

**Comment by akshika47** [comment]
@zhanwenchen Thank you. This helped me.

**Comment by andreaBelmont** [comment]
hello, any chance that we can see this bug fixed?

**Comment by akshika47** [comment]
Good point @andreaBelmont. I am happy to help as well.

**Comment by drskuce** [comment]
I tried to hack this file on mac for latest swi 8.0.3 but after 1 hour I decided I didnt want to get into this

could someone fix this so us mac people can get back to work??

thanks

DS
